### PR TITLE
Standard deviation challenge

### PR DESCRIPTION
**Pull request by Adamx090**
---
Name: Solution Review for the Standard Deviation Challenge #45 (includes commits from upstream)
About: A template PR for code review with a checklist
---

### Files

- [x] The file name describes the function's behavior
- [x] There is a module docstring in the function file
- [x] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [x] There is a module docstring in the tests file

### Unit Tests

- [x] The test class has a helpful name in PascalCase
- [x] The test class has a docstring
- Every unit test has
  - [x] A helpful name
  - [x] A clear docstring
  - [x] Only one assertion
  - [x] There is no logic in the unit test
- [x] All tests pass
- [x] There are tests for defensive assertions
- [x] There are tests for boundary cases

### Function Docstring

- [x] The function's behavior is described
- The function's arguments are described:
  - [x] Type
  - [x] Purpose
  - [x] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [x] Type
  - [x] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [x] Each assumption about an argument is checked with an assertion
  - [x] Each assertion checks for _only one_ assumption about the argument
- [x] Include 3 or more (passing!) doctests

### The Function

- [x] The function's name describes it's behavior
- [x] The function's name matches the file name
- [x] The function has correct type annotations

### Do's

- [x] Variable names help to understand the strategy
- [x] Any comments are clear and describe the strategy
- [x] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [x] The function's strategy _is not_ described in the documentation
- [x] Comments explain the _strategy_, **not** the _implementation_
- [x] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [x] The code passes the formatting checks
- [x] The code passes all Ruff linting checks
- [x] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [x] Variables are named with snake_case
- [x] Variable names are clear and helpful
- [x] The code follows the strategy as simply as possible
- [x] The implementation is as simple as possible given the strategy
- [x] The code includes defensive assertions
- [x] Defensive assertions include as little logic as possible

### CODE REVIEW

File: solutions/tests/test_standard_deviation.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+    def test_empty_list_input(self):
+        """It should raise AssertionError if input is an empty list"""
+        with self.assertRaises(AssertionError):
+            standard_deviation([])
+
+    def test_list_items_not_integers_or_floats(self):
+        """It should raise AssertionError if list items are not integers or float"""
+        with self.assertRaises(AssertionError):
+            standard_deviation([55, "-222", 177.65, "thirty", 86])
```
[the remark this answers is not in the document]
Comment (Adamx090) [reply solutions/tests/test_standard_deviation.py:85]: Good point. I haven't used append() in my function, but I should get into the habit of always testing for side effects when dealing with data structures like lists.
Already added that to my unit tests.

**Review by AlhassenSabeeh — APPROVED** [review]
